app/over_react_redux/todo_client/test/unit/browser/redux/store_test.dart

```diff
@@ -0,0 +1,452 @@
+            reason: reasonCurrentSetShouldBePersisted);
+      });
+    });
+
+    group('selectedTodosReducer updates state', () {
+      void select() {
+        final initialSelectedTodoIds = testStore.state.selectedTodoIds;
+        expect(initialSelectedTodoIds, isEmpty, reason: 'test setup sanity check');
+        final newSelectedTodoId = testStore.state.todos.first.id;
+        var expectedNewState = [newSelectedTodoId];
+        testStore.dispatch(SelectTodoAction(newSelectedTodoId));
+
+        expect(testStore.state.selectedTodoIds, expectedNewState);
+        expect(getCurrentLocalStorageSet()['selectedTodoIds'], expectedNewState,
+            reason: reasonCurrentSetShouldBePersisted);
+
+        final anotherNewSelectedTodoId = testStore.state.todos[1].id;
+        expectedNewState = [newSelectedTodoId, anotherNewSelectedTodoId];
+        testStore.dispatch(SelectTodoAction(anotherNewSelectedTodoId));
+
+        expect(testStore.state.selectedTodoIds, expectedNewState);
+        expect(getCurrentLocalStorageSet()['selectedTodoIds'], expectedNewState,
+            reason: reasonCurrentSetShouldBePersisted);
+      }
+
+      test('when an SelectTodoAction is dispatched', select);
+
+      test('when a DeselectTodoAction is dispatched', () {
+        select();
+
+        final noLongerSelectedTodoId = testStore.state.todos.first.id;
+        testStore.dispatch(DeselectTodoAction(noLongerSelectedTodoId));
+
+        expect(testStore.state.selectedTodoIds, isNot(contains(noLongerSelectedTodoId)));
+        expect(getCurrentLocalStorageSet()['selectedTodoIds'], isNot(contains(noLongerSelectedTodoId)),
+            reason: reasonCurrentSetShouldBePersisted);
+      });
+    });
+
+    group('highlightedTodosReducer updates state', () {
+      void highlight() {
+        final initialHighlightedTodoIds = testStore.state.highlightedTodoIds;
+        expect(initialHighlightedTodoIds, isEmpty, reason: 'test setup sanity check');
+        var expectedNewState = [
+          testStore.state.todos[0].id,
+          testStore.state.todos[1].id,
+        ];
+        testStore.dispatch(HighlightTodosAction(expectedNewState));
+
+        expect(testStore.state.highlightedTodoIds, expectedNewState);
+        expect(getCurrentLocalStorageSet()['highlightedTodoIds'], expectedNewState,
+            reason: reasonCurrentSetShouldBePersisted);
+
+        final anotherNewHighlightedTodoId = testStore.state.todos[2].id;
+        expectedNewState.add(anotherNewHighlightedTodoId);
+        testStore.dispatch(HighlightTodosAction([anotherNewHighlightedTodoId]));
+
+        expect(testStore.state.highlightedTodoIds, expectedNewState);
+        expect(getCurrentLocalStorageSet()['highlightedTodoIds'], expectedNewState,
+            reason: reasonCurrentSetShouldBePersisted);
+      }
+
+      test('when an HighlightTodosAction is dispatched', highlight);
+
+      test('when a UnHighlightTodosAction is dispatched', () {
```

Comment:
#nit I think `an` is proper here? 🤷‍♂ 
```suggestion
      test('when an UnHighlightTodosAction is dispatched', () {
```